client/tests/integration/multisig.rs

```diff
@@ -0,0 +1,170 @@
+use std::{
+    collections::{BTreeMap, BTreeSet},
+    str::FromStr,
+};
+
+use eyre::Result;
+use iroha::{
+    client::{self, ClientQueryError},
+    crypto::KeyPair,
+    data_model::{
+        account::NewAccount,
+        prelude::*,
+        transaction::{TransactionBuilder, WasmSmartContract},
+    },
+};
+use iroha_data_model::parameter::SmartContractParameter;
+use nonzero_ext::nonzero;
+use serde::{Deserialize, Serialize};
+use test_network::*;
+use test_samples::{gen_account_in, ALICE_ID};
+
+/// Mirror trigger args for `multisig_register`
+#[derive(Serialize, Deserialize)]
+struct MultisigRegisterArgs {
+    // Account id of multisig account should be manually checked to not have corresponding private key (or having master key is ok)
+    account: NewAccount,
+    // List of accounts responsible for handling multisig account
+    signatories: BTreeSet<AccountId>,
+}
+
+/// Mirror trigger args for `multisig`
+#[derive(Serialize, Deserialize)]
+enum MultisigArgs {
+    /// Accept instruction and vote
+    Instruction(InstructionBox),
+    /// Accept vote for certain instruction
+    Vote(HashOf<InstructionBox>),
+}
+
+#[test]
+fn mutlisig() -> Result<()> {
+    let (_rt, _peer, test_client) = <PeerBuilder>::new().with_port(11_400).start_with_runtime();
+    wait_for_genesis_committed(&vec![test_client.clone()], 0);
+
+    test_client.submit_all_blocking([
+        SetParameter::new(Parameter::SmartContract(SmartContractParameter::Fuel(
+            nonzero!(100_000_000_u64),
+        ))),
+        SetParameter::new(Parameter::Executor(SmartContractParameter::Fuel(nonzero!(
+            100_000_000_u64
+        )))),
+    ])?;
+
+    let account_id = ALICE_ID.clone();
+    let multisig_register_trigger_id = TriggerId::from_str("multisig_register")?;
+
+    let wasm =
+        iroha_wasm_builder::Builder::new("tests/integration/smartcontracts/multisig_register")
+            .show_output()
+            .build()?
+            .optimize()?
+            .into_bytes()?;
+    let wasm = WasmSmartContract::from_compiled(wasm);
+
+    let trigger = Trigger::new(
+        multisig_register_trigger_id.clone(),
+        Action::new(
+            wasm.clone(),
+            Repeats::Indefinitely,
+            account_id.clone(),
+            ExecuteTriggerEventFilter::new().for_trigger(multisig_register_trigger_id.clone()),
+        ),
+    );
+
+    // Register trigger which would allow multisig account creation in wonderland domain
+    // Access to call this trigger shouldn't be restricted
+    test_client.submit_blocking(Register::trigger(trigger))?;
+
+    // Create multisig account id and destroy it's private key
+    let multisig_account_id = gen_account_in("wonderland").0;
```

Comment:
- this should be a variant of `AccountId` (essentially a variant of `PublicKey`) to reject transactions that claim to have this multisig authority, which could be left to another issue
- since multisig account ID is not associated with its signatories, they would need some way to know that the multisig account is theirs

Reply:
- the former holds if multisig account is created on user side as this
  - if created on system side, such a case could be statistically ignored that user restores the private key and correctly signs
- the latter holds if created on system side with some tweak with account registration

Reply:
> since multisig account ID is not associated with its signatories, they would need some way to know that the multisig account is theirs

My thinking was that creation of multisig account is always initialized as user action (submitting transaction) so interested party already know that they are signatories of a given account.

Reply:
> this should be a variant of AccountId (essentially a variant of PublicKey) to reject transactions that claim to have this multisig authority, which could be left to another issue

There is might be valid use cases to have multisig for regular workflow and private key as backup solution i.e. regular workers have to gather votes and boss can use private key in case of any problem. 

Alternatively signatories might derive public key as sum of theirs public keys (or generated by them for this particular multisig account) so unless every signatory would have to share there private keys with each other otherwise it should be infeasible to find aggregated private key.

Reply:
I see, it makes sense to me to leave it up to user to decide whether or not to have the corresponding private key